File: backend/src/main/java/com/devmatch/backend/domain/analysis/repository/AnalysisRepository.java
package com.devmatch.backend.domain.analysis.repository;

import com.devmatch.backend.domain.analysis.entity.AnalysisResult;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface AnalysisRepository extends JpaRepository<AnalysisResult, Long> {

  Optional<AnalysisResult> findByApplicationId(Long applicationId);
}

File: backend/src/main/java/com/devmatch/backend/domain/analysis/repository/AnalysisRepository.kt
package com.devmatch.backend.domain.analysis.repository

import com.devmatch.backend.domain.analysis.entity.AnalysisResult
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface AnalysisRepository : JpaRepository<AnalysisResult, Long> {
    fun findByApplicationId(applicationId: Long): AnalysisResult?
}